src/Castle.Facilities.AspNetCore/CastleWindsorRegistrationExtensions.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright 2004-2018 Castle Project - http://www.castleproject.org/
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.Facilities.AspNetCore
+{
+	using System;
+	using System.Linq;
+
+	using Castle.MicroKernel.Lifestyle;
+	using Castle.MicroKernel.Registration;
+	using Castle.Windsor;
+
+	using Microsoft.AspNetCore.Builder;
+	using Microsoft.AspNetCore.Http;
+	using Microsoft.AspNetCore.Mvc;
+	using Microsoft.AspNetCore.Razor.TagHelpers;
+	using Microsoft.Extensions.DependencyInjection;
+
+	public static class CastleWindsorRegistrationExtensions
+	{
+		private static readonly string[] InvalidFrameworkConfigurationNamespaces =
+		{
+			"Microsoft.AspNetCore"
+			// Todo: Might need to add some more framework namespaces.
+		};
+
+		public static void AddCastleWindsor(this IServiceCollection services, IWindsorContainer container)
+		{
+			services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
```

Comment:
I don't see `IHttpContextAccessor` used in the facility, just wanting to understand what it is needed for.